storm/src/main/storm/mesos/NimbusMesosScheduler.java

```diff
@@ -109,6 +119,48 @@ public void statusUpdate(SchedulerDriver driver, TaskStatus status) {
     }
   }
 
+  private void updateLogviewerState(TaskStatus status) {
+    String taskId = status.getTaskId().getValue();
+    if (!taskId.contains(MesosCommon.MESOS_COMPONENT_ID_DELIMITER)) {
+      LOG.error("updateLogviewerState: taskId for logviewer, {}, isn't formatted correctly", taskId);
+      return;
+    }
+    String nodeId = taskId.split("\\" + MesosCommon.MESOS_COMPONENT_ID_DELIMITER)[1];
+    String logviewerZKPath = String.format("%s/%s", logviewerZkDir, nodeId);
+    switch (status.getState()) {
+      case TASK_STAGING:
+        checkRunningLogviewerState(logviewerZKPath);
+        return;
+      case TASK_STARTING:
+        checkRunningLogviewerState(logviewerZKPath);
+        return;
+      case TASK_RUNNING:
+        checkRunningLogviewerState(logviewerZKPath);
+        return;
+      case TASK_LOST:
+        // this status update can be triggered by the explicit kill and isn't terminal, do not kill again
+        break;
+      default:
+        // explicitly kill the logviewer task to ensure logviewer is terminated
+        mesosNimbus._driver.killTask(status.getTaskId());
```

Comment:
Now I'm a bit scared of this behavior. :)  Hopefully we won't have a combination of logviewers failing to be able to launch along with a cyclical task update behavior like we saw (i.e., receive TASK_LOST -> call killTask -> receive another TASK_LOST -> ...).

@changreytang @JessicaLHartog what is your feeling about this?

Reply:
@erikdw Personally I don't think the cycle would happen again after we handled the TASK_LOST specifically. The use case for this line of code is that we get a non-terminal task death update (caused by a network partition) in which the logviewer is actually still running so we kill it explicitly such that we won't accidentally launch two logviewers on the same host. 

SIDE NOTE: I think a good check to eventually put in before launching a logviewer on a host is to check that the logviewer port is free. Currently, the logviewer process itself doesn't care if the port is already taken and still runs, which is the reason the logviewer process flood is able to happen. If we put a check of the port before launching a logviewer on a host I think that would be helpful check for ensuring that only one logviewer is running on each host.

Reply:
@changreytang : yup, I heard what you are saying about checking the logviewer port.  I'm not sure how to do that nicely, maybe a wrapper / initial cmd to be executed which needs to succeed before we launch the logviewer.  Where that cmd could be something *like*: 

```
% sudo netstat -anpl | grep -e ':8080\s\s*.*LISTEN'
tcp        0      0 :::8080                     :::*                        LISTEN      22916/java  
```

Reply:
Issue #218 filed